main/src/mill/main/MillServerMain.scala

```diff
@@ -134,12 +135,26 @@ class Server[T](
     }.getOrElse(throw new Exception("PID already present"))
   }
 
+  def proxyInputStreamThroughPumper(in: InputStream) = {
+    val pipedInput = new PipedInputStream()
+    val pipedOutput = new PipedOutputStream()
+    pipedOutput.connect(pipedInput)
+    val pumper = new InputPumper(in, pipedOutput, false)
+    val pumperThread = new Thread(pumper)
```

Comment:
This thread should be marked as daemon via `pumperThread.setDaemon(true)`, I think.

Reply:
Possibly? Things seem to work fine without it tho, because in `mill.MillMain.main` we explicitly call `System.exit` at the end

Reply:
Yeah, I think if we agree its nature _is_ daemon, it's no problem to also mark it as such. It may make probably also our own lives easier, eventually.

Motivation:
I collected a long history of attempts to use apps as libraries. Relying on `System.exit` is the first and foremost blocker to do that. In my integrator-role I often searched for alternative entry points to tools without `System.exit`. But this always assumes, that the rest of the resources are well managed.

Reply:
Got it. Added the `setDaemon` call

Reply:
Thanks!